nx/lib/nx.ex

```diff
@@ -6747,6 +6747,131 @@ defmodule Nx do
     |> divide(weights_sum)
   end
 
+  @doc """
+  Returns the median for the tensor.
+
+  If the `:axis` option is given, it aggregates over
+  that dimension, effectively removing it. `axis: 0`
+  implies aggregating over the highest order dimension
+  and so forth. If the axis is negative, then the axis will
+  be counted from the back. For example, `axis: -1` will
+  always aggregate over the last dimension.
+
+  You may optionally set `:keep_axis` to true, which will
+  retain the rank of the input tensor by setting the reduced
+  axis to size 1.
+
+  ## Examples
+
+      iex> Nx.median(Nx.tensor(42))
+      #Nx.Tensor<
+        s64
+        42
+      >
+
+      iex> Nx.median(Nx.tensor([1, 2, 3]))
+      #Nx.Tensor<
+        s64
+        2
+      >
+
+      iex> Nx.median(Nx.tensor([1, 2]))
+      #Nx.Tensor<
+        f32
+        1.5
+      >
+
+  ### Aggregating over an axis
+
+      iex> Nx.median(Nx.tensor([[1, 2, 3], [4, 5, 6]], names: [:x, :y]), axis: 0)
+      #Nx.Tensor<
+        f32[y: 3]
+        [2.5, 3.5, 4.5]
+      >
+
+      iex> Nx.median(Nx.tensor([[1, 2, 3], [4, 5, 6]], names: [:x, :y]), axis: :y)
+      #Nx.Tensor<
+        s64[x: 2]
+        [2, 5]
+      >
+
+      iex> t = Nx.tensor([[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]], names: [:x, :y, :z])
+      iex> weights = Nx.tensor([[[0, 1, 2], [1, 1, 0]], [[-1, 1, -1], [1, 1, -1]]])
+      iex> Nx.weighted_mean(t, weights, axis: :x)
+      #Nx.Tensor<
+        f32[y: 2][z: 3]
+        [
+          [7.0, 5.0, -3.0],
+          [7.0, 8.0, 12.0]
+        ]
+      >
+
+      iex> t = Nx.tensor([[[1, 2, 2], [3, 4, 2]], [[4, 5, 2], [7, 9, 2]]])
+      iex> Nx.median(t, axis: -1)
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [2, 3],
+          [4, 7]
+        ]
+      >
+
+  ### Keeping axis
+
+      iex> t = Nx.tensor([[[1, 2, 2], [3, 4, 2]], [[4, 5, 2], [7, 9, 2]]])
```

Comment:
I would be nice to add a test (not a doctesr, though) where the input tensor is not sorted